src/equations/shallow_water_3d.jl

```diff
@@ -0,0 +1,333 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+@doc raw"""
+    ShallowWaterEquations3D(; gravity, H0 = 0)
+
+Shallow water equations (SWE) in three space dimensions in conservation form (with constant bottom topography). 
+The equations are given by
+```math
+\begin{aligned}
```

Comment:
My understanding is that the 3D Cartesian form of the shallow water equations first appeared in the following paper, which we should reference here:

J. Coté (1988). A Lagrange multiplier approach for the metric terms of semi-Lagrangian models on the sphere. Quarterly Journal of the Royal Meteorological Society 114, 1347-1352. https://doi.org/10.1002/qj.49711448310

I also think it would make sense to mention the Lagrange multiplier source term, and perhaps link to where that is implemented (i.e. in the elixir).

Reply:
Thanks. I moved the Lagrange multiplier techniques, improved the description of these methods, and added the reference that you proposed in 8a9ef3d.